### ISSUE TITLE
test-title rule should ignore tests that use macros

### ISSUE THREAD

**Issue by novemberborn** [avajs/eslint-plugin-ava#168]
Just ran into the macro issue I was referring to in https://github.com/avajs/ava/issues/1207#issuecomment-275187528. Take this macro:

```
const asIs = (t, value) => {
  t.true(unwrapSync(value) === value)
}
asIs.title = (kind, value, note) => `returns ${kind} values ${note ? `(${note}) ` : ''}as-is`

test('object', asIs, {}, 'without .then')
test('object', asIs, { then: true }, 'without .then being a function')
```

This fails our ESLint rule because the title is repeated. Arguably this would make more sense:

```
const asIs = (t, value) => {
  t.true(unwrapSync(value) === value)
}
asIs.title = (title, value, kind, note) => `returns ${kind} values ${note ? `(${note}) ` : ''}as-is`

test(asIs, {}, 'object', 'without .then')
test(asIs, { then: true }, 'object', 'without .then being a function')
```

But with https://github.com/avajs/ava/issues/1207 in mind, ESLint will soon fail because there is no title. 

The `test-title` rule should detect both scenarios and ignore them.

**Comment by jfmengels** [comment]
This sounds a lot more like the `no-identical-title` rule than `test-title`. I made a fix for the former, but if this is indeed the latter, let me know.

**Comment by novemberborn** [comment]
@jfmengels yes, `no-identical-title`. These rules have very ~identical~ similar titles 😉 .
